smart-contract/assembly/contracts/deweb-interface.ts

```diff
@@ -1,187 +1,369 @@
+ * Retrieves a specific chunk of a file.
+ * @param _binaryArgs - Serialized FileChunkGet object containing hashLocation and index.
+ * @returns The requested chunk as a StaticArray<u8>.
+ * @throws If the chunk request is invalid.
+ */
+export function getFileChunk(_binaryArgs: StaticArray<u8>): StaticArray<u8> {
+  const args = new Args(_binaryArgs);
+  const chunk = args.next<FileChunkGet>().expect('Invalid chunk');
+  return _getFileChunk(chunk.hashLocation, chunk.index);
+}
+
+export function removeFileChunkRange(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+  const args = new Args(_binaryArgs);
+  const hashLocation = args
+    .next<StaticArray<u8>>()
+    .expect('Invalid hashLocation');
+
+  const start = args.next<u32>().expect('Invalid start');
+  const end = args.next<u32>().expect('Invalid end');
+
+  _removeChunksRange(hashLocation, start, end);
+}
+
+/**
+ * Retrieves the total number of chunks for a file.
+ * @param _binaryArgs - Serialized FileChunkGet object containing hashLocation.
+ * @returns The total number of chunks.
+ * @throws If the hashLocation is invalid.
+ */
+export function getFileChunkCount(
+  _binaryArgs: StaticArray<u8>,
+): StaticArray<u8> {
+  const args = new Args(_binaryArgs);
+  const hashLocation = args
+    .next<StaticArray<u8>>()
+    .expect('Invalid hashLocation');
+
+  return u32ToBytes(_getTotalChunk(hashLocation));
+}
+
+/**
+ * Deletes files from the contract storage.
  * Only the contract owner can call this function.
- * @param _binaryArgs - Serialized arguments containing an array of PreStore objects.
- * @throws If the preStore data is invalid or if the caller is not the owner.
+ * @param _binaryArgs - Serialized arguments containing an array of FileDelete objects.
+ * @throws If the files are invalid or if the caller is not the owner.
  */
-export function preStoreFileChunks(_binaryArgs: StaticArray<u8>): void {
+export function deleteFiles(_binaryArgs: StaticArray<u8>): void {
   _onlyOwner();
+
   const args = new Args(_binaryArgs);
   const files = args
-    .nextSerializableObjectArray<PreStore>()
-    .expect('Invalid preStore');
+    .nextSerializableObjectArray<FileDelete>()
+    .expect('Invalid files');
 
   for (let i = 0; i < files.length; i++) {
-    const totalChunks = _getTotalChunk(files[i].filePathHash);
-
-    if (totalChunks === 0) {
-      _pushFilePath(files[i].filePath);
-    }
-
-    // Remove the file if the new total chunks is 0
-    if (files[i].newTotalChunks === 0) {
-      _removeFile(files[i].filePath, files[i].filePathHash, totalChunks - 1);
-    } else {
-      if (files[i].newTotalChunks < totalChunks) {
-        _removeChunksRange(
-          files[i].filePathHash,
-          files[i].newTotalChunks - 1,
-          totalChunks - 1,
-        );
-      }
-      _setTotalChunk(files[i].filePathHash, files[i].newTotalChunks);
-    }
+    assert(_getTotalChunk(files[i].hashLocation) > 0, 'File does not exist');
+
+    _deleteFile(files[i].hashLocation);
```

Comment:
we should discuss metadata support for files contained inside subdirectories within ZIP files. Currently, the SC assumes that only files with available chunks can have metadata. Would be nice to be able to set metadata and location - but not chunks (eg. chunk_count entry absent) - for some files.

Reply:
As it's not a requirement for the v1 use case, can it be discussed for a v2 ?

Reply:
okay, maybe let's open an issue to not forget :)

Reply:
Sure